client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -120,6 +120,24 @@ const DomainPicker: FunctionComponent< Props > = ( {
 		if ( allSuggestions ) {
 			setRailcarId( getNewRailcarId() );
 		}
+
+		if ( freeSuggestions && recommendedSuggestion && paidSuggestions ) {
+			if (
+				currentDomain?.is_free === true &&
+				currentDomain?.domain_name !== freeSuggestions[ 0 ].domain_name
+			) {
+				onDomainSelect( freeSuggestions[ 0 ] );
+			}
+
+			if (
+				currentDomain?.is_free !== true &&
+				! paidSuggestions.find(
+					( suggestion ) => suggestion.domain_name === currentDomain?.domain_name
+				)
+			) {
+				onDomainSelect( recommendedSuggestion );
+			}
+		}
 	}, [ allSuggestions ] );
```

Comment:
There's an eslint warning here about missing dependencies. Maybe having a seperate `useEffect()` would help, but `freeSuggestions`, etc. should be listed too.